<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Overview of the iSEE package • iSEEWorkshopEuroBioc2020</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Overview of the iSEE package">
<meta property="og:description" content="iSEEWorkshopEuroBioc2020">
<meta property="og:image" content="https://iSEE.github.io/iSEEWorkshopEuroBioc2020/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">iSEEWorkshopEuroBioc2020</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/overview.html">Overview of the iSEE package</a>
    </li>
    <li>
      <a href="../articles/dataset.html">Preprocessing a data set for iSEE</a>
    </li>
    <li>
      <a href="../articles/recipes.html">Hands-on workshop recipes</a>
    </li>
  </ul>
</li>
<li>
  <a href="https://github.com/iSEE/iSEEWorkshopEuroBioc2020/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="overview_files/accessible-code-block-0.0.1/empty-anchor.js"></script><link href="overview_files/font-awesome-5.1.0/css/all.css" rel="stylesheet">
<link href="overview_files/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Overview of the iSEE package</h1>
                        <h4 class="author">Kevin Rue-Albrecht, Charlotte Soneson, Federico Marini, and Aaron Lun</h4>
            
            <h4 class="date">December 15, 2020</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/iSEE/iSEEWorkshopEuroBioc2020/blob/master/vignettes/overview.Rmd"><code>vignettes/overview.Rmd</code></a></small>
      <div class="hidden name"><code>overview.Rmd</code></div>

    </div>

    
    
<!-- Enable snowflakes in vignettes


<div class="snowflakes" aria-hidden="true">
  <div class="snowflake">
  ❅
  </div>
  <div class="snowflake">
  ❅
  </div>
  <div class="snowflake">
  ❆
  </div>
  <div class="snowflake">
  ❄
  </div>
  <div class="snowflake">
  ❅
  </div>
  <div class="snowflake">
  ❆
  </div>
  <div class="snowflake">
  ❄
  </div>
  <div class="snowflake">
  ❅
  </div>
  <div class="snowflake">
  ❆
  </div>
  <div class="snowflake">
  ❄
  </div>
</div>
-->
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>This vignette provides an overview of the graphical interface of <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em> applications. To follow along, make sure that you launch the default <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em> instance as described in the next code block at the start of the <a href="#default-app">following section</a>.</p>
<p>Note that in the default configuration, the panels do not look exactly like the ones shown in the screenshots that you will see below. For example, data points are not immediately colored, and the default annotation variables displayed by each panel may differ. The <a href="recipes.html">hands-on recipes</a> workshop of this workshop will demonstrate how to modify the content of the panels, and how they are displayed.</p>
<p>Note that for simplicity, we typically refer to a <code>SummarizedExperiment</code> in this workshop; however, <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em> works seamlessly for objects of any class extending <code>SummarizedExperiment</code> as well (e.g., <code>SingleCellExperiment</code>, <code>DESeqDataSet</code>). That said, some types of panels – such as the Reduced dimension plot – are only available for objects that contain a <code>reducedDim</code> slot (in particular, <code>SingleCellExperiment</code> objects); the basic <code>SummarizedExperiment</code> class does not contain this slot. In this workshop, we refer to the rows of the <code>SummarizedExperiment</code> object as ‘features’ (these can be genes, transcripts, genomic regions, etc) and to the columns as ‘samples’ (which, in our example data set, are single cells).</p>
</div>
<div id="default-app" class="section level1">
<h1 class="hasAnchor">
<a href="#default-app" class="anchor"></a>The default iSEE app</h1>
<p>Using the demonstration data set, we can launch an <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em> instance for exploring this data set using the <code><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html">iSEE()</a></code> function without any further argument. This will produce an app using the default configuration; that is, the app instance will include one panel of each built-in class for which the relevant information is available in the <code>SummarizedExperiment</code> object.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/iSEE/iSEE">iSEE</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://iSEE.github.io/iSEEWorkshopEuroBioc2020">iSEEWorkshopEuroBioc2020</a></span><span class="op">)</span>
<span class="va">sce</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/load_demo_sce.html">load_demo_sce</a></span><span class="op">(</span><span class="st">"pbmc3k"</span><span class="op">)</span>
<span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html">iSEE</a></span><span class="op">(</span><span class="va">sce</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://shiny.rstudio.com/reference/shiny/latest/runApp.html">runApp</a></span><span class="op">(</span><span class="va">app</span><span class="op">)</span></code></pre></div>
<p><img src="screenshots/isee-initial.png" width="100%"></p>
<p>The main argument to the <code><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html">iSEE()</a></code> function is a <code>SummarizedExperiment</code> object, or an object of any class extending <code>SummarizedExperiment</code> (such as <code>SingleCellExperiment</code>, in this case). No other restrictions are made on the type of data stored in the object, and <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em> can be used for interactive visualization of many different types of data. It is also worth noting that for various types of data, Bioconductor packages provides functionality for directly importing quantifications generated by external software packages into a <code>SummarizedExperiment</code> object. For example, the <em><a href="https://bioconductor.org/packages/3.13/DropletUtils">DropletUtils</a></em> package can read quantifications from the 10x Genomics CellRanger pipeline for single-cell RNA-seq data, and the <em><a href="https://bioconductor.org/packages/3.13/tximeta">tximeta</a></em> package can be used to read data from transcript quantification pipelines into a <code>SummarizedExperiment</code> object.</p>
<p>While we will not make explicit use of it in this workshop, we note that it is also possible to call <code><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html">iSEE()</a></code> without providing a <code>SummarizedExperiment</code> object. In that case, the user will be prompted to upload such an object, serialized into a <code>.rds</code> file. It is also possible to import a specification of the initial panel setup.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html">iSEE</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://shiny.rstudio.com/reference/shiny/latest/runApp.html">runApp</a></span><span class="op">(</span><span class="va">app</span><span class="op">)</span></code></pre></div>
<p><img src="screenshots/isee-no-se.png" width="100%"></p>
</div>
<div id="user-interface" class="section level1">
<h1 class="hasAnchor">
<a href="#user-interface" class="anchor"></a>The built-in panel types</h1>
<div id="the-panel-hierarchy" class="section level2">
<h2 class="hasAnchor">
<a href="#the-panel-hierarchy" class="anchor"></a>The panel hierarchy</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em> user interface (UI) consists of a number of panels, each displaying the data provided in the <code>SummarizedExperiment</code> from a specific perspective. There are 8 standard panel types; 6 plot panels and 2 table panels, all showcased in the figure shown at the start of the <a href="#default-app">previous section</a>. In the default configuration, one panel of each type is included when launching the <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em> user interface. However, users are free to rearrange, resize, remove or add panels freely, as demonstrated in the separate vignette of <a href="recipes.html">workshop recipes</a>. We provide a brief overview of each panel type in the following subsections.</p>
<p>In addition to the 8 standard panel types included in <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em>, users can create custom panels (both plots and tables). Moreover, the <em><a href="https://bioconductor.org/packages/3.13/iSEEu">iSEEu</a></em> (iSEE universe) package contains additional panel types. The creation and configuration of custom panels is also demonstrated in a series of <a href="recipes.html">workshop advanced recipes</a>.</p>
<p>Specifically, <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em> panels are implemented as a hierarchy of S4 classes. Only <em>concrete</em> classes (orange) may be instantiated as panels in the user interface; while <em>virtual</em> classes (grey) provide functionality shared by families of panels (e.g., sample-oriented plots, feature-oriented tables).</p>
<p><img src="schematics/panel-classes.png" width="100%"></p>
<!--
This is perhaps not very pedagogical - the app will not look like the one below if it is just opened with the object created above.
But it's nice to show the capabilities of the panels too, not just the default settings.
I added a few sentences to the introduction of this section to try to address this.
-->
</div>
<div id="reddimplot" class="section level2">
<h2 class="hasAnchor">
<a href="#reddimplot" class="anchor"></a>Reduced dimension plot</h2>
<p>The reduced dimension plot can display any reduced dimension representation that is present in the <code>reducedDim</code> slot of the <code>SingleCellExperiment</code> object.</p>
<p>Note that this slot is not defined for the base <code>SummarizedExperiment</code> class, in which case the user interface does not allow the inclusion of panels of this type.</p>
<p><img src="screenshots/isee-reduceddimensionplot.png" width="100%"></p>
</div>
<div id="coldataplot" class="section level2">
<h2 class="hasAnchor">
<a href="#coldataplot" class="anchor"></a>Column data plot</h2>
<p>The column data plot can display one or two of the provided column annotations (from the <code>colData</code> slot). Depending on the class of the selected annotations, the panel shows either a scatter plot, a violin plot, or a Hinton diagram <span class="citation">(Hinton and Shallice 1991; Bremner, Gotts, and Denham 1994)</span>.</p>
<p><img src="screenshots/isee-columndataplot.png" width="100%"></p>
</div>
<div id="rowdataplot" class="section level2">
<h2 class="hasAnchor">
<a href="#rowdataplot" class="anchor"></a>Row data plot</h2>
<p>Analogous to the <a href="#coldataplot">column data plot</a> above, the row data plot displays one or two of the provided row annotations (from the <code>rowData</code> slot). Depending on the class of the selected annotations, the panel displays either a scatter plot, a violin plot, or a Hinton plot.</p>
<p><img src="screenshots/isee-rowdataplot.png" width="100%"></p>
</div>
<div id="heatmap" class="section level2">
<h2 class="hasAnchor">
<a href="#heatmap" class="anchor"></a>Complex heatmap</h2>
<p>The complex heatmap panel displays, for any assay, the observed values for a subset of the features across the samples.</p>
<p><img src="screenshots/isee-complexheatmapplot.png" width="100%"></p>
</div>
<div id="featassayplot" class="section level2">
<h2 class="hasAnchor">
<a href="#featassayplot" class="anchor"></a>Feature assay plot</h2>
<p>The feature assay plot displays the observed values for one feature across the samples. It is also possible to plot the observed values for two features, in a scatter plot.</p>
<p><img src="screenshots/isee-featureassayplot.png" width="100%"></p>
</div>
<div id="sampassayplot" class="section level2">
<h2 class="hasAnchor">
<a href="#sampassayplot" class="anchor"></a>Sample assay plot</h2>
<p>Analogous to the <a href="#featassayplot">Feature assay plot</a> above, the Sample assay plot shows the observed values for all features, for one of the samples. It is also possible to plot the observed values for two samples, in a scatter plot.</p>
<p><img src="screenshots/isee-sampleassayplot.png" width="100%"></p>
</div>
<div id="rowdatatable" class="section level2">
<h2 class="hasAnchor">
<a href="#rowdatatable" class="anchor"></a>Row data table</h2>
<p>The row data table displays all information provided in the <code>rowData</code> slot of the <code>SummarizedExperiment</code> object, leveraging the interactivity provided by the <em><a href="https://CRAN.R-project.org/package=DT">DT</a></em> package.</p>
<p><img src="screenshots/isee-rowdatatable.png" width="100%"></p>
</div>
<div id="coldatatable" class="section level2">
<h2 class="hasAnchor">
<a href="#coldatatable" class="anchor"></a>Column data table</h2>
<p>Analogous to the <a href="#rowdatatable">Row data table</a> above, the column data table displays all information provided in the <code>colData</code> slot of the <code>SummarizedExperiment</code> object.</p>
<p><img src="screenshots/isee-columndatatable.png" width="100%"></p>
</div>
</div>
<div id="collapsible-boxes" class="section level1">
<h1 class="hasAnchor">
<a href="#collapsible-boxes" class="anchor"></a>Collapsible boxes with display controls</h1>
<div id="data-parameters-box" class="section level2">
<h2 class="hasAnchor">
<a href="#data-parameters-box" class="anchor"></a>Data parameters</h2>
<p>Each plot panel type has a <code>Data parameters</code> collapsible box. This box has different content for each panel type, but in all cases it lets the user control the data that is displayed in the plot.</p>
<p><img src="screenshots/isee-columndataplot-dataparameters.png" width="100%"></p>
</div>
<div id="visual-parameters-box" class="section level2">
<h2 class="hasAnchor">
<a href="#visual-parameters-box" class="anchor"></a>Visual parameters</h2>
<p>In contrast to the <code>Data parameters</code> collapsible box that lets users control <em>what</em> is displayed in the plot, the <code>Visual parameters</code> box lets users control <em>how</em> the information is displayed.</p>
<p>This collapsible box contains the controls to change the size, shape, opacity, and color of the points, to facet the plot by any available categorical annotation, to subsample points for increased speed of plot rendering, and to control how legends are displayed.</p>
<p><img src="screenshots/isee-columndataplot-visualparameters.png" width="100%"></p>
</div>
<div id="selection-parameter-box" class="section level2">
<h2 class="hasAnchor">
<a href="#selection-parameter-box" class="anchor"></a>Selection parameters</h2>
<p>The <code>Selection parameters</code> collapsible box provides controls to transfer selections of points (features or samples) between panels.</p>
<p>We demonstrate examples of point transmission in the separate vignette of <a href="recipes.html">workshop recipes</a></p>
<p><img src="screenshots/isee-reduceddimensionplot-selectionparameters.png" width="100%"></p>
</div>
</div>
<div id="additional-controls" class="section level1">
<h1 class="hasAnchor">
<a href="#additional-controls" class="anchor"></a>Additional controls</h1>
<p>At the top-right corner of the iSEE application, users can find additional controls for reproducibility, configuration, and help.</p>
<table class="table"><tbody>
<tr class="odd">
<td>Organization</td>
<td align="center"><img src="controls/PanelOrganization.png" width="433"></td>
<td>
<a href="#organize-panels">Organize panels</a> <br><a href="#examine-panel-chart">Examine panel chart</a>
</td>
</tr>
<tr class="even">
<td>Export</td>
<td align="center"><img src="controls/Export.png" width="386"></td>
<td>
<a href="#download-output">Download panel output</a> <br><a href="#extract-r-code">Extract R code</a> <br><a href="#display-panel-settings">Display panel settings</a>
</td>
</tr>
<tr class="odd">
<td>Documentation</td>
<td align="center"><img src="controls/Documentation.png" width="337"></td>
<td>
<a href="#quick-tour">Quick tour</a> <br><a href="#browse-vignettes">Open the vignette</a>
</td>
</tr>
<tr class="even">
<td>Additional info</td>
<td align="center"><img src="controls/AdditionalInfo.png" width="298"></td>
<td>
<a href="#app-session-info">About this session</a> <br><a href="#about-isee">About iSEE</a>
</td>
</tr>
</tbody></table>
<div id="organize-panels" class="section level2">
<h2 class="hasAnchor">
<a href="#organize-panels" class="anchor"></a>Organize panels</h2>
<p>As mentioned above, the default behaviour of the <code><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html">iSEE()</a></code> function is to launch an instance of the user interface that displays one panel of each of the standard types (provided the underlying data is available, e.g. for reduced dimension plots the <code>reducedDim</code> slot is required). However, in some cases it is desirable to have multiple panels of the same type, and/or exclude some panel types.</p>
<p>In order to accommodate such situations, users can add, remove, change the order of and resize all panels via the <code>Organization</code> menu in the top-right corner.</p>
<p>Clicking in the selectize box listing all current panels will present you with a drop-down menu from which you can choose additional panels to add. Similarly, panels can be removed by clicking on the <i class="fa fa-times"></i> icon associated with the panel name.</p>
<p>Each panel can be individually resized by changing the width and height. Note that the total width of a row in the interface is 12 units. When the width of a panel is greater than the space available, the panel is moved to a new row.</p>
<p><img src="modals/panel-organization.png" width="100%"></p>
</div>
<div id="examine-panel-chart" class="section level2">
<h2 class="hasAnchor">
<a href="#examine-panel-chart" class="anchor"></a>Linking panels and transmitting point selections</h2>
<p>When exploring data, it is often useful to be able to select a subset of points and investigate different aspects of their characteristic features. In <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em>, this can be achieved by <em>selecting</em> points in one panel, and <em>transmitting</em> this selection to one or more other panels.</p>
<p>The brushing and point selection can also be programmatically preconfigured.</p>
<p>The button <code>Examine panel chart</code> display a graph that reports any active point transmission between panels in the app.</p>
<p><img src="modals/panel-graph.png" width="100%"></p>
</div>
<div id="download-output" class="section level2">
<h2 class="hasAnchor">
<a href="#download-output" class="anchor"></a>Download panel output</h2>
<p>The <code>Download panel output</code> button opens a modal window listing all the current panels in the app. Checkboxes allow users to select any subset of panels to export. Finally, clicking the <code>Download</code> button in that modal will prompt the app to save plots to PDF files, tables to CSV files, package the set of files in a ZIP archive that users can download and save on their computer.</p>
<p><img src="modals/panel-download.png" width="100%"></p>
</div>
<div id="extract-r-code" class="section level2">
<h2 class="hasAnchor">
<a href="#extract-r-code" class="anchor"></a>Extract R code</h2>
<p>The fact that data exploration is done interactively is no reason to forego reproducibility! To this end, <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em> lets you export the exact R code used to create each of the currently visible plots.</p>
<p>Importantly, the script reported by <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em> contains a short preamble needed to set up variable names that are used in individual panels, including active brushes used to transfer point selections.</p>
<p><img src="modals/code-tracker.png" width="100%"></p>
</div>
<div id="display-panel-settings" class="section level2">
<h2 class="hasAnchor">
<a href="#display-panel-settings" class="anchor"></a>Display panel settings</h2>
<p>It can take a great amount of time to achieve a satisfactory panel configuration. To avoid the need to manually organize the panels each time the app is opened, <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em> offers the possibility to export code that can be reused later to programmatically specify how the app is initialized, as well as to inspect and export the current panel settings for future use.</p>
<p><img src="modals/panel-settings.png" width="100%"></p>
</div>
<div id="quick-tour" class="section level2">
<h2 class="hasAnchor">
<a href="#quick-tour" class="anchor"></a>Quick tour</h2>
<p>One important aspect of visualization is the ability to share your insights with others. A powerful way of easily getting people unfamiliar with your data up to speed is to provide a walkthrough of the interface and the different types of plots that are displayed. With <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em>, this can be achieved using <em>tours</em>. To configure a tour, you need to create a text file with two columns; named <code>element</code> and <code>intro</code>, with the former describing the UI element to highlight in each step, and the latter providing the descriptive text that will be displayed.</p>
<p>Clicking the <code>Quick tour</code> button launches an interactive tour of the interface using the <em><a href="https://CRAN.R-project.org/package=rintrojs">rintrojs</a></em> package, highlighting specific elements of the user interface, labeled with information and instructions guiding users through panels and tasks specific to individual apps.</p>
<p><img src="modals/quick-tour.png" width="100%"></p>
<p>You can see a live example of a more complex tour in action, visit <a href="https://marionilab.cruk.cam.ac.uk/iSEE_pbmc4k/" class="uri">https://marionilab.cruk.cam.ac.uk/iSEE_pbmc4k/</a>. The code used to create this app and the associated tour is available from <a href="https://github.com/iSEE/iSEE2018/tree/master/tours/pbmc4k" class="uri">https://github.com/iSEE/iSEE2018/tree/master/tours/pbmc4k</a>.</p>
</div>
<div id="browse-vignettes" class="section level2">
<h2 class="hasAnchor">
<a href="#browse-vignettes" class="anchor"></a>Open the vignette</h2>
<p>The user interface navigation bar also includes this button to open the introductory vignette to <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em> in your we browser.</p>
<p>Depending on the version of <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em> that you are using, this will adaptively lead you to a locally built vignette present on your computer, or the <em>release</em> or the <em>devel</em> version of the Bioconductor package landing page, e.g. &lt;<a href="https://bioconductor.org/packages/3.13/iSEE/vignettes/basic.html">Bioconductor 3.13</a>&gt;.</p>
</div>
<div id="app-session-info" class="section level2">
<h2 class="hasAnchor">
<a href="#app-session-info" class="anchor"></a>Session information</h2>
<p>This button displays the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo()</a></code>, which is a useful piece of information to report when reporting an issue with an app.</p>
<p><img src="modals/session-info.png" width="100%"></p>
</div>
<div id="about-isee" class="section level2">
<h2 class="hasAnchor">
<a href="#about-isee" class="anchor"></a>About iSEE</h2>
<p>This button provides information about the authors of <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em>, as well as citation information.</p>
<p>If you use this package, please use the following citation information:</p>
<blockquote>
<p>Rue-Albrecht K, Marini F, Soneson C, Lun ATL (2018). “iSEE: Interactive SummarizedExperiment Explorer.” <em>F1000Research</em>, <em>7</em>, 741. doi: 10.12688/f1000research.14966.1 (URL: <a href="https://doi.org/10.12688/f1000research.14966.1" class="uri">https://doi.org/10.12688/f1000research.14966.1</a>).</p>
</blockquote>
<p>A BibTeX entry for LaTeX users is:</p>
<pre><code>@Article{,
  title = {iSEE: Interactive SummarizedExperiment Explorer},
  author = {Kevin Rue-Albrecht and Federico Marini and Charlotte Soneson and Aaron T. L. Lun},
  publisher = {F1000 Research, Ltd.},
  journal = {F1000Research},
  year = {2018},
  month = {Jun},
  volume = {7},
  pages = {741},
  doi = {10.12688/f1000research.14966.1},
}</code></pre>
</div>
</div>
<div id="additional-resources" class="section level1">
<h1 class="hasAnchor">
<a href="#additional-resources" class="anchor"></a>Additional resources</h1>
<p>A lot of materials have been developed, from the first use of <em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em> to contributions of new panel classes in separate downstream packages.</p>
<ul>
<li>The bookdown book on <em>Extending iSEE</em>: <a href="https://isee.github.io/iSEE-book/" class="uri">https://isee.github.io/iSEE-book/</a>
</li>
<li>Bioconductor landing page: <a href="https://bioconductor.org/packages/iSEE/" class="uri">https://bioconductor.org/packages/iSEE/</a>
</li>
<li>Publication (F1000Research, 2018): <a href="https://f1000research.com/articles/7-741/v1" class="uri">https://f1000research.com/articles/7-741/v1</a>
</li>
<li>Deployed examples: <a href="https://marionilab.cruk.cam.ac.uk/" class="uri">https://marionilab.cruk.cam.ac.uk/</a>, code at <a href="https://github.com/iSEE/iSEE2018" class="uri">https://github.com/iSEE/iSEE2018</a>
<ul>
<li>Further deployments in the <a href="https://github.com/iSEE/iSEE_instances" class="uri">https://github.com/iSEE/iSEE_instances</a> repo</li>
<li>
<em><a href="https://bioconductor.org/packages/3.13/iSEE">iSEE</a></em> in production: <a href="http://www.teichlab.org/singlecell-treg" class="uri">http://www.teichlab.org/singlecell-treg</a>
</li>
</ul>
</li>
<li>Custom panel examples: <a href="https://github.com/iSEE/iSEE_custom" class="uri">https://github.com/iSEE/iSEE_custom</a>
</li>
<li>Development version (bug reports etc): <a href="https://github.com/iSEE/iSEE" class="uri">https://github.com/iSEE/iSEE</a>
</li>
</ul>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Bremner1994-hintonplots">
<p>Bremner, Frederick J, Stephen J Gotts, and Dina L Denham. 1994. “Hinton Diagrams: Viewing Connection Strengths in Neural Networks.” <em>Behav. Res. Methods Instrum. Comput.</em> 26 (2): 215–18.</p>
</div>
<div id="ref-Hinton1991-hintondiagram">
<p>Hinton, Geoffrey E, and Tim Shallice. 1991. “Lesioning an Attractor Network: Investigations of Acquired Dyslexia.” <em>Psychological Review</em> 98 (1): 74–95.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Kevin Rue-Albrecht, Federico Marini, Charlotte Soneson, Aaron Lun.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
